core/src/main/scala/org/apache/spark/scheduler/JobWaiter.scala

```diff
@@ -17,6 +17,10 @@
 
 package org.apache.spark.scheduler
 
+import sun.misc.{Signal, SignalHandler}
```

Comment:
@hvanhovell thank you very much for your comment. I understand your concern.

But Apache Hive and Facebook Presto both import ``sun.misc.Signal/SignalHander`` [here](https://github.com/apache/hive/blob/master/cli/src/java/org/apache/hadoop/hive/cli/CliDriver.java#L80) and [here](https://github.com/facebook/presto/blob/master/presto-cli/src/main/java/com/facebook/presto/cli/Query.java#L29). I am positive they're heavily tested against any other JDKs.

In addition, JLine implementation that you're referring to also use the same package via reflection in the end. All it does is adding some safety guard for the case ``sun.misc.Signal/SignalHander`` might not exist, which is unlikely in any major JDKs.